src/components/FilterPanel.jsx

```diff
@@ -18,38 +20,66 @@ import CloseIcon from '@mui/icons-material/Close';
 import PropTypes from 'prop-types';
 
 export default function FilterPanel({ onClose }) {
-  const [{ minAmount, maxAmount, grades, majors }, setQueryParam] =
+  const [{ minAmount, maxAmount, grades, majors }, setQueryParams] =
     useQueryParams();
+  const [filterVals, setFilterVals] = useState({
+    minAmount,
+    maxAmount,
+    grades,
+    majors,
+  });
 
   const filters = {
-    Major: (
-      <MajorFilter
-        majors={majors}
-        onSelect={(m) => setQueryParam('majors', m)}
-        onDelete={(m) =>
-          setQueryParam(
-            'majors',
-            majors.filter((major) => major !== m)
-          )
-        }
-      />
-    ),
-    Amount: (
-      <AmountFilter
-        min={minAmount ?? 0}
-        max={maxAmount ?? 0}
-        onMinChange={(val) => setQueryParam('minAmount', val)}
-        onMaxChange={(val) => setQueryParam('maxAmount', val)}
-      />
-    ),
-    'Grade Level': (
-      <GradeLevelFilter
-        grades={new Set(grades)}
-        changeFn={(e) => setQueryParam('grades', e)}
-      />
-    ),
+    Major: {
+      comp: (
+        <MajorFilter
+          majors={filterVals.majors}
+          onSelect={(m) => setFilterVals({ ...filterVals, majors: m })}
+          onDelete={(m) =>
+            setFilterVals({
+              ...filterVals,
+              majors: filterVals.majors.filter((major) => major !== m),
+            })
+          }
+        />
+      ),
+      badge: majors?.length,
+    },
+    Amount: {
+      comp: (
+        <AmountFilter
+          min={filterVals.minAmount ?? 0}
+          max={filterVals.maxAmount ?? 0}
+          onMinChange={(val) =>
+            setFilterVals({ ...filterVals, minAmount: parseInt(val) })
+          }
+          onMaxChange={(val) =>
+            setFilterVals({ ...filterVals, maxAmount: parseInt(val) })
+          }
+        />
+      ),
+      badge: minAmount && maxAmount ? 2 : minAmount || maxAmount ? 1 : 0,
+    },
+    'Grade Level': {
+      comp: (
+        <GradeLevelFilter
+          grades={new Set(filterVals.grades)}
+          changeFn={(e) => setFilterVals({ ...filterVals, grades: e })}
+        />
+      ),
+      badge: grades?.length,
+    },
   };
 
+  function resetFilters() {
+    setFilterVals({
+      minAmount: undefined,
+      maxAmount: undefined,
+      grades: undefined,
+      majors: undefined,
```

Comment:
set to `undefined` so they can be pruned. I could also write a separate function that clears the filters, that way I can set this to `{}`